docs: explain --take-ownership for install and upgrade (#2207)
helm/helm-www: docs/topics/advanced.mdx

For more information on using the Go SDK, See the [Go SDK section](#go-sdk)

> ## Taking ownership of existing resources

Reviewer: ```suggestion
## Allow Helm to take ownership of existing resources
```
Just to be a bit more specific about who/what would be taking ownership of those resources

Reviewer: Note: Promptless called out using title case instead of sentence case, but this page is already inconsistent about that, so I'm not worried about for this PR.

> By default, Helm refuses to install or upgrade a chart if a target resource already exists in the cluster and is not already owned by the release. Ownership is tracked with Helm-managed annotations (and related labels such as `app.kubernetes.io/managed-by`). That check protects you from clobbering objects that another tool or release created.

Reviewer: ```suggestion
By default, Helm refuses to install or upgrade a chart if a resource that is not owned by the release already exists in the cluster.
In this case, Helm would display the following error: "resource already exists and is not owned by this release".

Helm tracks resource ownership in the cluster with both Helm-managed annotations and related labels such as `app.kubernetes.io/managed-by`.
This ownership validation process protects you from unintentionally overwriting objects that another tool or release had created.
```
Copyedits for clarity/style, moves up the failure message users might see, and adds semantic line breaks

Use `--take-ownership` on `helm install` or `helm upgrade` when you intentionally want Helm to adopt those existing objects into the release:

```shell
helm install my-release ./mychart --take-ownership
helm upgrade my-release ./mychart --take-ownership
```

> With the flag set, Helm skips the "resource already exists and is not owned by this release" failure, adopts the live objects, and applies the chart's desired state. Subsequent upgrades and uninstalls then treat those objects as part of the release.

Reviewer: ```suggestion
With the flag set, Helm ignores the resource ownership checks and adopts the existing objects into the release.
Subsequent upgrades and uninstalls then treat those objects as part of the release.
```

> ### When to use it

Reviewer: ```suggestion
Common use cases for the `--take-ownership` flag include:
```

- Migrating a manually applied manifest (or another deployer) to a Helm chart without deleting the live resources first
- Recovering after annotations were removed or a release history was lost, so Helm no longer recognizes its own objects
- Aligning a chart with cluster objects that already match the desired names/kinds

> ### Caveats

Reviewer: ```suggestion
### Limitations and best practices
```

> - Adoption does **not** delete extra fields that were never in the chart. Three-way strategic merge (or server-side apply, if you use it) still controls how live and chart values are combined.

Reviewer: ```suggestion
- Taking ownership doesn't delete extra fields that were never in the chart. Three-way strategic merge (or server-side apply, if you use it) still controls how live and chart values are combined.
```

> - Taking ownership of the wrong objects can surprise other operators that still expect to manage them. Prefer matching names carefully and reviewing the dry-run output first (`--dry-run`).

Reviewer: ```suggestion
- To avoid taking ownership of objects that Helm shouldn't manage, match names carefully and review the dry-run output first (`--dry-run`).
```
